airflow/models/dagrun.py

```diff
@@ -486,10 +515,25 @@ def get_task_instances(
                 else:
                     tis = tis.where(TI.state.in_(state))
 
-        if self.dag and self.dag.partial:
-            tis = tis.where(TI.task_id.in_(self.dag.task_ids))
+        if dag and dag.partial:
+            tis = tis.where(TI.task_id.in_(dag.task_ids))
         return session.scalars(tis).all()
 
+    @provide_session
+    def get_task_instances(
+        self,
+        state: Iterable[TaskInstanceState | None] | None = None,
+        session: Session = NEW_SESSION,
+    ) -> list[TI]:
+        """
+        Returns the task instances for this dag run.
+        Redirect to DagRun.fetch_task_instances method.
+        Keep this method because it is widely used across the code.
+        """
```

Comment:
Please revise docstrings to conform to D205. http://www.pydocstyle.org/en/2.1.1/error_codes.html